File: 1. Arrays & Strings/5_pair_sum/5_pair_sum.java
import java.util.List;
import java.util.ArrayList;
import java.util.HashMap;

class Source {
    public static List<Integer> pairSum(List<Integer> numbers, int target) {
        List<Integer> result = new ArrayList<>();
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < numbers.size(); i++) {
            int newTarget = target - numbers.get(i);

            if (map.containsKey(newTarget)) {
                result = List.of(map.get(newTarget), i);
                break;
            } else {
                map.put(numbers.get(i), i);
            }
        }

        return result;
    }
}

File: 1. Arrays & Strings/5_pair_sum/5_pair_sum.py
def pair_sum(numbers, target):
    result = []
    num_map = {}

    for i, number in enumerate(numbers):
        new_target = target - number

        if new_target in num_map:
            result = [num_map[new_target], i]
            break
        else:
            num_map[number] = i

    return result
